test/functional/interface_zmq_dash.py

```diff
@@ -238,6 +289,28 @@ def test_instantsend_publishers(self):
         zmq_double_spend_tx_1.deserialize(self.receive(ZMQPublisher.raw_instantsend_doublespend))
         assert(zmq_double_spend_tx_1.is_valid())
         assert_equal(zmq_double_spend_tx_1.hash, rpc_raw_tx_1['txid'])
+        # No islock notifications when tx is not received yet
+        self.nodes[0].generate(1)
+        rpc_raw_tx_3 = self.create_raw_tx(self.nodes[0], self.nodes[0], 1, 1, 100)
+        islock = self.create_islock(rpc_raw_tx_3['hex'])
+        self.test_node.send_islock(islock)
+        # Validate NO hashtxlock
+        time.sleep(1)
+        try:
+            data = self.receive(ZMQPublisher.hash_tx_lock, zmq.NOBLOCK)
```

Comment:
Wouldn't it make sense to do something like this, so that if there is a tx lock, the test fails?

```suggestion
            data = self.receive(ZMQPublisher.hash_tx_lock, zmq.NOBLOCK)
            assert(false)
```